source/slang/slang-ir-lower-bit-cast.cpp

```diff
@@ -238,6 +247,10 @@ struct BitCastLoweringContext
         {
             return;
         }
+
+        if (fromTypeSize.size != toTypeSize.size)
```

Comment:
Shouldn't we also check the alignment?
For the primitive types, the alignments are same as their sizes but for structs, the alignments are the biggest alignment value among the members. That case give us a case where the alignments are different when their sizes are same.